diff -Naur uhttpd-2014-10-27/client.c ../uhttpd-2014-10-27/client.c
--- uhttpd-2014-10-27/client.c	2016-05-09 10:04:35.000000000 +0800
+++ ../uhttpd-2014-10-27/client.c	2018-06-12 10:15:33.000000000 +0800
@@ -53,7 +53,9 @@
 	if (!uh_use_chunked(cl))
 		enc = "";
 
-	if (r->connection_close)
+	if (cl->tls == TRUE) {
+		conn = "Connection: close";
+	} else if (r->connection_close)
 		conn = "Connection: close";
 	else
 		conn = "Connection: Keep-Alive";
diff -Naur uhttpd-2014-10-27/file.c ../uhttpd-2014-10-27/file.c
--- uhttpd-2014-10-27/file.c	2016-05-09 10:04:35.000000000 +0800
+++ ../uhttpd-2014-10-27/file.c	2018-06-12 10:17:43.000000000 +0800
@@ -814,7 +814,12 @@
 {
 	struct http_request *req = &cl->request;
 	struct dispatch_handler *d;
-	char *url = blobmsg_data(blob_data(cl->hdr.head));
+	char *url = NULL;
+	if (cl->tls == true)
+		url = "/portal/portal_redirect.html";
+	else {
+		url = blobmsg_data(blob_data(cl->hdr.head));
+	}
 	char *error_handler;
 
 	req->redirect_status = 200;
